### PR TITLE
8253680: Vector API VectorMask and VectorSpecies tests for coverage

### PR DESCRIPTION
**Pull request by sviswa7**
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [JDK-8253680](https://bugs.openjdk.java.net/browse/JDK-8253680): Vector API VectorMask and VectorSpecies tests for coverage

### Reviewers
 * [Ningsheng Jian](https://openjdk.java.net/census#njian) (@nsjian - Committer) ⚠️ Review applies to eb3866c693fcc09dc737bbcb32f2434f1815f961
 * [Paul Sandoz](https://openjdk.java.net/census#psandoz) (@PaulSandoz - Committer)

### Download
`$ git fetch https://git.openjdk.java.net/panama-vector pull/9/head:pull/9`
`$ git checkout pull/9`

### CODE REVIEW

**Comment by sviswa7** [comment]
This patch adds the following tests to increase code coverage:
 VectorMask methods (eq, equals, fromLong, hashCode, indexInRange, lastTrue, toLong, trueCount)
 VectorSpecies methods (loopBound)

**Review by nsjian — APPROVED** [review]
Looks good to me and tested on AArch64.

File: test/jdk/jdk/incubator/vector/templates/Unit-Miscellaneous.template [hunk 1]
```diff
@@ -364,6 +364,155 @@
...
+    static void maskTrueCount$vectorteststype$SmokeTest(IntFunction<boolean[]> fa) {
+        boolean[] a = fa.apply(SPECIES.length());
+
+        for (int i = 0; i < a.length; i += SPECIES.length()) {
+            var vmask = SPECIES.loadMask(a, i);
+            int tcount = vmask.trueCount();
+#if[!MaxBit]
+            int expectedTcount = Long.bitCount(vmask.toLong());
+#else[!MaxBit]
+            int expectedTcount = 0;
```
Comment (PaulSandoz) [inline test/jdk/jdk/incubator/vector/templates/Unit-Miscellaneous.template:424]: I think you should consistently derive the expected count from the source `a` regardless of the kind of vector.

File: test/jdk/jdk/incubator/vector/templates/Unit-Miscellaneous.template [hunk 2]
```diff
@@ -364,6 +364,155 @@
...
+                {0x5555555555555555l},
+                {0x0123456789abcdefl},
+        };
+    }
+
+    @Test(dataProvider = "longMaskProvider")
+    static void maskFromToLong$vectorteststype$SmokeTest(long inputLong) {
+        var vmask = VectorMask.fromLong(SPECIES, inputLong);
+        long outputLong = vmask.toLong();
+        Assert.assertEquals(outputLong, inputLong & (((1l << (SPECIES.length()-1)) << 1)-1));
```
Comment (PaulSandoz) [inline test/jdk/jdk/incubator/vector/templates/Unit-Miscellaneous.template:467]: Recommend using `L` for the identifier of a long value (see also array declaration in `longMaskProvider`)
since lower case letter `l` is hard to distinguish from the digit `1`, and placing spaces in the subtracting expression. Alternatively could use `SPECIES.maskAll(true).toLong()` to avoid a complex expression dealing with the edge case of 64 lanes.

File: test/jdk/jdk/incubator/vector/templates/Unit-Miscellaneous.template [hunk 3]
```diff
@@ -364,6 +364,155 @@
...
+                {-2},
+        };
+    }
+
+    @Test(dataProvider = "offsetProvider")
+    static void indexInRange$vectorteststype$SmokeTest(int offset) {
+        int limit = SPECIES.length() * BUFFER_REPS;
+        for (int i = 0; i < limit; i += SPECIES.length()) {
+            var actualMask = SPECIES.indexInRange(i+offset, limit);
+            var expectedMask = SPECIES.maskAll(true).indexInRange(i+offset, limit);
```
Comment (PaulSandoz) [inline test/jdk/jdk/incubator/vector/templates/Unit-Miscellaneous.template:487]: `i+offset` -> `i + offset` and for other expressions

**Review by PaulSandoz — CHANGES_REQUESTED** [review]

File: test/jdk/jdk/incubator/vector/templates/Unit-Miscellaneous.template (hunk 1 above)
Comment (sviswa7) [reply test/jdk/jdk/incubator/vector/templates/Unit-Miscellaneous.template:421]: Done.

File: test/jdk/jdk/incubator/vector/templates/Unit-Miscellaneous.template (hunk 3 above)
Comment (sviswa7) [reply test/jdk/jdk/incubator/vector/templates/Unit-Miscellaneous.template]: Done.

File: test/jdk/jdk/incubator/vector/templates/Unit-Miscellaneous.template (hunk 2 above)
Comment (sviswa7) [reply test/jdk/jdk/incubator/vector/templates/Unit-Miscellaneous.template]: Added L and missing spaces. Kept the complex expression as the test itself is for fromLong and toLong so using toLong in expectedValue may not be preferable.

**Review by PaulSandoz — APPROVED** [review]
